src/s1reader/s1_annotation.py

```diff
@@ -0,0 +1,497 @@
+            for i,elem in enumerate(element_to_parse):
+                str_elem = elem.find(name_vector).text
+                list_out[i] = np.array([int(strin) for strin in str_elem.split()])
+
+        elif str_type == 'vector_float':
+            for i,elem in enumerate(element_to_parse):
+                str_elem = elem.find(name_vector).text
+                list_out[i] = np.array([float(strin) for strin in str_elem.split()])
+
+        elif str_type == 'str':
+            list_out = element_to_parse[0].find(name_vector).text
+
+        else:
+            raise ValueError(f'Cannot recognize the type of the element: {str_type}')
+
+        return list_out
+
+
+@dataclass
+class CalibrationAnnotation(AnnotationBase):
+    '''Reader for Calibration Annotation Data Set (CADS)'''
+    list_azimuth_time: np.ndarray
+    list_line: list
+    list_pixel: None
+    list_sigma_nought: list
+    list_beta_nought : list
+    list_gamma: list
+    list_dn: list
+
+    @classmethod
+    def from_et(cls, et_in=None):
+        '''Extracts the list of calibration informaton from etree from CADS'''
+        cls.xml_et = et_in
+        cls.list_azimuth_time = cls._parse_vectorlist('calibrationVectorList', 'azimuthTime', 'datetime')
+        cls.list_line = cls._parse_vectorlist('calibrationVectorList', 'line', 'scalar_int')
+        cls.list_pixel = cls._parse_vectorlist('calibrationVectorList', 'pixel', 'vector_int')
+        cls.list_sigma_nought = cls._parse_vectorlist('calibrationVectorList', 'sigmaNought', 'vector_float')
+        cls.list_beta_nought = cls._parse_vectorlist('calibrationVectorList', 'betaNought', 'vector_float')
+        cls.list_gamma = cls._parse_vectorlist('calibrationVectorList', 'gamma', 'vector_float')
+        cls.list_dn = cls._parse_vectorlist('calibrationVectorList', 'dn', 'vector_float')
+
+        return cls
+
+
+
+
+@dataclass
+class NoiseAnnotation(AnnotationBase):
+    '''Reader for Noise Annotation Data Set (NADS) for IW SLC'''
+    # NOTE Schema of the NADS is slightly different before/after IPF version 2.90. Needs to be adaptive in accordance with the version.
+    # The issue above was fixed in further implement of thermal noise correction. A separete PR regarding this will be submitted upon the acceptance of this code.
+    # in ISCE2 code: if float(self.IPFversion) < 2.90:
+    # REF: .../isce2/components/isceobj/Sensor/GRD/Sentinel1.py
+
+    rg_list_azimuth_time: np.ndarray
+    rg_list_line: list
+    rg_list_pixel: list
+    rg_list_noise_range_lut: list
+    az_first_azimuth_line: int
+    az_first_range_sample: int
+    az_last_azimuth_line: int
+    az_last_range_sample: int
+    az_line: np.ndarray
+    az_noise_azimuth_lut: np.ndarray
+
+    @classmethod
+    def from_et(cls,et_in: ET, ipf_version: version.Version = version.parse('3.10')):
+        '''Extracts list of noise information from etree'''
+        
+        if et_in is not None:
+            cls.xml_et = et_in
```

Comment:
Thanks for the suggestion. I think the current code is not efficient in dealing with unacceptable `et_in`.
I've removed the line you pointed. I might need to think about the data type of `et_in` is not acceptable (i.e. not ET), but I think it can be handled by the function header.

```python
@classmethod
    def from_et(cls,et_in: ET, ipf_version: version.Version):
```

Also please note that the default value in the parameter was removed as per your [suggestion above.](https://github.com/opera-adt/s1-reader/pull/48#discussion_r935134460)